x-pack/plugins/security/server/saved_objects/saved_objects_security_extension.ts

```diff
@@ -139,33 +229,114 @@ export class SavedObjectsSecurityExtension implements ISavedObjectsSecurityExten
     return { typeMap, status: 'unauthorized' };
   }
 
+  private auditHelper(
+    action: AuditAction,
+    objects?: Array<{ type: string; id: string }>,
+    useSuccessOutcome: boolean = false,
+    addToSpaces?: string[],
+    deleteFromSpaces?: string[],
+    error?: Error
+  ) {
+    if (objects && objects?.length > 0) {
```

Comment:
__nit:__ would be great to have a comment explaining the `else` branch too. Optionally we can join them with something like this (assuming they stay almost identical, but up to you):
```ts
// We should also log audit event even if we don't have SO because ....
const savedObjects = objects && objects?.length > 0 ? objects : ([undefined] as undefined[]);
for (const obj of savedObjects) {
  this.addAuditEvent({
    savedObject: obj && { type: obj.type, id: obj.id },
```

Reply:
I updated the the auditHelper with a variation of your suggestion. Thanks!